### PR TITLE
Migration guide for deprecated identifiers.

### PR DESCRIPTION
**Pull request by evan-masseau**
Add migration guide entry for the deprecated `ProfileKey` objects

### CODE REVIEW

**Review by ajaysubra — APPROVED** [review]

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -2,6 +2,40 @@
 This document provides guidance on how to migrate from the old version of the SDK to a newer version. 
 It will be updated as new versions are released including deprecations or breaking changes.
 
+## 2.1.0 Deprecations
```
Comment (kennyklaviyo) [inline MIGRATION_GUIDE.md:5]: is this the right version?

**Review by jordan-griffin — APPROVED** [review]

File: MIGRATION_GUIDE.md (hunk 1 above)
Comment (evan-masseau) [reply MIGRATION_GUIDE.md:5]: nope

Comment (evan-masseau) [reply MIGRATION_GUIDE.md:5]:
```suggestion
## 2.3.0 Deprecations
```

Comment (evan-masseau) [reply MIGRATION_GUIDE.md:5]: thank youuu

**Pull request merged**